hw_4_flask.py

```diff
@@ -0,0 +1,82 @@
+import random
+
+
+import requests
+import pandas as pd
+
+from datetime import datetime
+from flask import Flask, jsonify, abort
+from faker import Faker
+from webargs import fields, validate
+from webargs.flaskparser import use_kwargs
+
+fake_instance = Faker('uk_UA')
+
+app = Flask(__name__)
+
+
+@app.route('/students')
+@use_kwargs(
+    {
+        'quantity': fields.Int(
+            load_default=50,
+            validate=[validate.Range(min=1, max=1000)]
+        )
+    },
+    location='query'
+)
+def generate_students(quantity):
+    students_data: list = [
+        {
+            'first_name': fake_instance.first_name(),
+            'last_name': fake_instance.last_name(),
+            'email': fake_instance.unique.email(),
+            'password': fake_instance.password(length=random.randint(10, 20)),
+            'birthday': str(fake_instance.date_of_birth(minimum_age=18, maximum_age=28))
+        } for data in range(quantity)
+    ]
+    df = pd.DataFrame(columns=['First Name', 'Last Name', 'Email', 'Password', 'Birthday'])
+    for p in students_data:
+        df = pd.concat([df, pd.DataFrame(p, index=[0])], ignore_index=True)
+        df.to_csv('students_data.csv', index=False)
+
+    app.config.update(JSON_SORT_KEYS=False)
+
+    return jsonify(students_data)
+
+
+def currency_checking(existing, request):
+    if request not in existing:
+        return abort(400, f'Incorrect currency code. Existing codes: {existing}')
```

Comment:
instead of abort better to use if statement and return error in view

Reply:
I don't understand what exactly you mean. To do a statement? Like this x = 1?

Reply:
Or do not make another func, and error statement in view?

Reply:
if currency not in currency_codes:
        abort(400, f'Incorrect currency code. Existing codes: {currency}')